pkg/registry/clusteroperator/clusterversion/storage_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterversion
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	genericregistrytest "k8s.io/apiserver/pkg/registry/generic/testing"
+	etcdtesting "k8s.io/apiserver/pkg/storage/etcd/testing"
+
+	clusteroperatorapi "github.com/openshift/cluster-operator/pkg/apis/clusteroperator"
+	"github.com/openshift/cluster-operator/pkg/registry/registrytest"
+)
+
+func newStorage(t *testing.T) (*genericregistry.Store, *etcdtesting.EtcdTestServer) {
+	etcdStorage, server := registrytest.NewEtcdStorage(t, clusteroperatorapi.GroupName)
+	restOptions := generic.RESTOptions{
+		StorageConfig:           etcdStorage,
+		Decorator:               generic.UndecoratedStorage,
+		DeleteCollectionWorkers: 1,
+		ResourcePrefix:          "clustersversions",
+	}
+	clusterVersionStorage, _ := NewStorage(restOptions)
+	return clusterVersionStorage, server
+}
+
+func validClusterVersion(name string) *clusteroperatorapi.ClusterVersion {
+	return &clusteroperatorapi.ClusterVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "helloworld",
```

Comment:
Do you mean to use the `name` parameter as the name of the `ClusterVersion`? Was this changed to "helloworld" to overcome "v3.9" not being a valid name?